### ISSUE TITLE
20.10: WIFI will not reconnect automatically.

### ISSUE THREAD

**Issue by ZeddieXX** [pop-os/pop#1490]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
N/A

**Issue/Bug Description:**
When laptop moves too far from WIFI AP or portable WIFI hotspot moves too far, Pop!_OS 20.10 will disconnect.  When laptop or hotspot moves within range, the WIFI does not reconnect as expected.  This happens with other laptops, so it is not a hardware specific issue.  The make/models of laptops I have tried this on:

System76 Galago Pro (GALP5) - Intel WIFI AX-201
HP zBook 15u G6 - Intel WIFI AX-200
HP EliteBook 840 G3 - Intel WIFI 8260 (rev 3a)
Dell Latitude E7450 - Intel WIFI 7265 (rev 59)
HP z400 (desktop) - Unknown USB Realtek-based WIFI (2.4 GHz only)

This happens on both the 2.4GHz and 5GHz bands, so it doesn't matter which one.  "Connect automatically" is enabled.

**Steps to reproduce (if you know):**
1. Make sure "Connect automatically" is enabled.  Connect to WIFI.  To ensure connectivity, ping a responding device on the network.
2. Take laptop (or WIFI hotspot) and walk away.  Ensure machine with Pop!_OS disconnects due to lost signal.
3. Walk laptop (or WIFI hotspot) closer.  You will notice machines with Pop!_OS installed will not reconnect unless you go back to the WIFI menu to pick a network to connect OR you turn WIFI off, then back on.
4. You will also notice other devices not running Pop!_OS (ex: Windows 10, EndevourOS, Arch) will reconnect once WIFI hotspot is in range.

**Expected behavior:**
Pop!_OS should reconnect to the WIFI signal as soon as it is within range.

**Other Notes:**
Connect automatically is enabled.  Doesn't matter the SSID name, 2.4 GHz, or 5 GHz.  Other OSes on the same hardware reconnects just fine when WIFI signal is back within range.

**Comment by 11allbits** [comment]
I can report that I have the same issue.  And it is a little more concerning - the wifi at my main place of work drops out for two minutes or so every few hours.  So, when the wifi returns, PopOS does not reconnect to the wifi automatically.  Every other operating system I have does reconnect automatically.  Therefore, PopOS becomes virtually unusable if I want to have services running in the background while I'm away from the machine.

**Comment by ZeddieXX** [comment]
I posted on PopOS subreddit and I didn't get many responses.  I am able to reproduce it with different hardware, so I thought for sure it would be affecting more people.

**Comment by curiousercreative** [comment]
@ZeddieXX just tested with mine and it reconnected! One thing that I changed on my pop install is I switched from the default network backend (networkd) to NetworkManager (because I handles wireguard routes from netplan config more intuitively). You can google around how to make the switch, but from my recollection, I had to:
1. Enable `NetworkManager` service
    ```
    sudo systemctl enable NetworkManger
    ```
2. Disable networkd
    ```
    sudo systemctl disable systemd-networkd
    ```
3. Reboot
    ```
    sudo systemctl reboot
    ```

To see if you're running NetworkManager already or to verify the switch: `sudo systemctl status NetworkManager`

**Comment by curiousercreative** [comment]
@ZeddieXX perhaps a safer way to test my hypothesis is to boot into a pop os recovery/live usb which is 20.04 and uses NetworkManager as network backend. Boot into that, connect to the wi-fi and see if it's improved. If it is, then consider booting back into 20.10 and trying to switch over permanently.

**Comment by ZeddieXX** [comment]
@curiousercreative Thanks.  I followed your steps to disable networkd and enable NetworkManager, but still having the same issue.

That was a good observation though.  I didn't know Pop!_OS 20.10 used networkd instead of NetworkManager.  I use NetworkManager in Arch, so that was a possible solution.

**Comment by ZeddieXX** [comment]
@curiousercreative Since switching to NetworkManager didn't help, I just switched back to systemd-networkd.

Now I don't see any NICs (wired or wireless).  Gnome Settings says NetworkManager needs to be running.

**Comment by curiousercreative** [comment]
@ZeddieXX that's a bummer, sorry to hear. I don't know exactly what to recommend other than to seek help on the pop os chat

**Comment by TC68** [comment]
Same issue here. I need to manually reconnect after a wifi dropout even though the connection is marked as "connect automatically".  I am on a Gazelle (gaze14) so do not expect a hardware issue.

NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os

All the online help I can find is for the case where it won't connect at all. In my case, connection happens automatically at boot as expected, but it just does not reconnect if the wifi signal drops out and returns

**Comment by ZeddieXX** [comment]
After not seeing any activity here from any devs who can reproduce the issue, I am thinking about moving to a different distro.

**Comment by curiousercreative** [comment]
@ZeddieXX did you already confirm via live boot with another distro that it's improved? I'd try with Ubuntu (maybe it's an upstream issue?) and something else. I just tested mine again and it reconnected immediately. It could also be network related? Does your network have multiple access points with the same SSID for example? I have just the one AP.

**Comment by TC68** [comment]
I haven't tracked it down - it is only a minor bother in my workflow. I'm not sure where the issue is. I mostly use gnome and the gui doesn't let you get to the same network settings as in Cinnamon so I may have something odd set. I have cinnamon as a backup DE and when I get a chance will test it out

**Comment by ZeddieXX** [comment]
> @ZeddieXX did you already confirm via live boot with another distro that it's improved? I'd try with Ubuntu (maybe it's an upstream issue?) and something else. I just tested mine again and it reconnected immediately. It could also be network related? Does your network have multiple access points with the same SSID for example? I have just the one AP.

I haven't tried a live Ubuntu, but will check it out now (downloading 20.10).  As for the SSID and APs, it happens at home (single WIFI router), 2 cellular hotspots (my phone, and my co-worker's phone), as well as my company's WIFI (AeroHive - 3 near me, same SSID of course - suppose to allow roaming).

ALL of them behave the same.  However, I'm now on Arch now (still fine tuning the install) and it works great.  It just reconnects without issues (on Gnome).

Downloading Ubuntu Live 20.10 now, so stay tuned.

**Comment by ZeddieXX** [comment]
FYI, also tried this on a Pop!_OS 20.10 live usb, and the behavior is the same.

**Comment by ZeddieXX** [comment]
Adding this update:  I have a ticket open with System76 since I had a GALP5 with Pop!_OS installed.

Aaron:
After sharing the new information provided in this support ticket our QA was finally able to recreate this issue. It seems to almost be a feature so that the system doesn't auto join a spotty, out of range network. Are you able to rejoin the network automatically once you turn off the AP and turn it back on?

ZeddieXX:
If I turn off the AP/Hotspot, then turn it back on, Pop!_OS 20.10 will still not reconnect.

This behavior isn't limited to a degrading WIFI signal. It also happens when there is a strong solid connection, then I turn off the hotspot. Later, when I turn it back on while still within range, it will NOT reconnect.

**Comment by jacobgkau** [comment]
I'm able to recreate this by turning a phone hotspot off and back on; however, I am also seeing the same behavior in a live disk of Ubuntu 20.10. The network reconnects the first time I turn the hotspot off and back on, but after that, it usually doesn't reconnect. (If I make a change to the configuration, such as saving a MAC address to the connection profile, it will reconnect a couple of times, then go back to not reconnecting.)

@ZeddieXX When you tested in Ubuntu 20.10 and saw different behavior than I'm seeing, how many times did you disconnect/reconnect? Are you waiting until the WiFi icon in the top-right menu goes away and the `Activation of network connection failed` notification appears before turning the hotspot back on? Logs for NetworkManager (`sudo journalctl -u NetworkManager | cat`) from Ubuntu re-connecting and Pop not re-connecting might be helpful to compare.

**Comment by jacobgkau** [comment]
Since you mentioned it was also working correctly in Arch, I tried a newer version of NetworkManager. (Ubuntu is currently shipping version 1.26.2, while Arch is shipping version 1.30.2.) Reconnection seems to be much more reliable with NetworkManager 1.30; I downloaded and installed the Debian Sid packages, but Ubuntu 21.04 (Hirsute) also has version 1.30 already.

If you'd like to test this out, download these eight Debian Sid .deb files:

https://packages.debian.org/sid/amd64/network-manager/download
https://packages.debian.org/sid/amd64/libnm0/download
https://packages.debian.org/sid/amd64/libgnutls30/download
https://packages.debian.org/sid/amd64/libnettle8/download
https://packages.debian.org/sid/amd64/network-manager-pptp/download
https://packages.debian.org/sid/amd64/network-manager-pptp-gnome/download
https://packages.debian.org/sid/amd64/ppp/download
https://packages.debian.org/sid/amd64/pptp-linux/download

Then install them using these commands:
```
cd Downloads
sudo dpkg -i ./network-manager_1.30.0-1_amd64.deb ./libgnutls30_3.7.1-1_amd64.deb ./libnm0_1.30.0-1_amd64.deb ./libnettle8_3.7-2.1_amd64.deb ./network-manager-pptp-gnome_1.2.8-3+b2_amd64.deb ./network-manager-pptp_1.2.8-3+b2_amd64.deb ./pptp-linux_1.10.0-1_amd64.deb ./ppp_2.4.9-1+1_amd64.deb
```

Then reboot. After that, do your networks reconnect the way you're expecting? (If so, I would recommend waiting until Ubuntu/Pop 21.04 are released later next month with the newer NetworkManager version.)

**Comment by ZeddieXX** [comment]
> 
> 
> I'm able to recreate this by turning a phone hotspot off and back on; however, I am also seeing the same behavior in a live disk of Ubuntu 20.10. The network reconnects the first time I turn the hotspot off and back on, but after that, it usually doesn't reconnect. (If I make a change to the configuration, such as saving a MAC address to the connection profile, it will reconnect a couple of times, then go back to not reconnecting.)
> 
> @ZeddieXX When you tested in Ubuntu 20.10 and saw different behavior than I'm seeing, how many times did you disconnect/reconnect? Are you waiting until the WiFi icon in the top-right menu goes away and the `Activation of network connection failed` notification appears before turning the hotspot back on? Logs for NetworkManager (`sudo journalctl -u NetworkManager | cat`) from Ubuntu re-connecting and Pop not re-connecting might be helpful to compare.

Honestly, I just leave with the hotspot for about 5 minutes then come back.  I noticed if I turn it off, then see the WiFi indicator on Gnome go away (shows disconnected), then turn it back on within a few seconds, it will reconnect just fine.  It's only after a few minutes of disconnection that seem to cause it to not reconnect when the hotspot comes back on.

Since I am now on Arch on my GALP5, I'm using a different laptop to troubleshoot this with you.  I'm on the HP zBook 15u G6.

When I have some time, I will cat the log for you and post it here.

**Comment by ZeddieXX** [comment]
> 
> 
> Since you mentioned it was also working correctly in Arch, I tried a newer version of NetworkManager. (Ubuntu is currently shipping version 1.26.2, while Arch is shipping version 1.30.2.) Reconnection seems to be much more reliable with NetworkManager 1.30; I downloaded and installed the Debian Sid packages, but Ubuntu 21.04 (Hirsute) also has version 1.30 already.
> 
> If you'd like to test this out, download these eight Debian Sid .deb files:
> 
> https://packages.debian.org/sid/amd64/network-manager/download
> https://packages.debian.org/sid/amd64/libnm0/download
> https://packages.debian.org/sid/amd64/libgnutls30/download
> https://packages.debian.org/sid/amd64/libnettle8/download
> https://packages.debian.org/sid/amd64/network-manager-pptp/download
> https://packages.debian.org/sid/amd64/network-manager-pptp-gnome/download
> https://packages.debian.org/sid/amd64/ppp/download
> https://packages.debian.org/sid/amd64/pptp-linux/download
> 
> Then install them using these commands:
> 
> ```
> cd Downloads
> sudo dpkg -i ./network-manager_1.30.0-1_amd64.deb ./libgnutls30_3.7.1-1_amd64.deb ./libnm0_1.30.0-1_amd64.deb ./libnettle8_3.7-2.1_amd64.deb ./network-manager-pptp-gnome_1.2.8-3+b2_amd64.deb ./network-manager-pptp_1.2.8-3+b2_amd64.deb ./pptp-linux_1.10.0-1_amd64.deb ./ppp_2.4.9-1+1_amd64.deb
> ```
> 
> Then reboot. After that, do your networks reconnect the way you're expecting? (If so, I would recommend waiting until Ubuntu/Pop 21.04 are released later next month with the newer NetworkManager version.)

When I get some time, I'll give this a try and report back.  I would LOVE to get back on Pop!_OS because I'm having issues with Arch not shutting down the GALP5 and I haven't had the time to troubleshoot it.  Other than this WIFI issue, I like using Pop!_OS as my main distro.

**Comment by ZeddieXX** [comment]
Here's the output
`zeddie@galp5:~/Downloads/network-manager$ sudo dpkg -i ./network-manager_1.30.0-1_amd64.deb ./libgnutls30_3.7.1-1_amd64.deb ./libnm0_1.30.0-1_amd64.deb ./libnettle8_3.7-2.1_amd64.deb ./network-manager-pptp-gnome_1.2.8-3+b2_amd64.deb ./network-manager-pptp_1.2.8-3+b2_amd64.deb ./pptp-linux_1.10.0-1_amd64.deb ./ppp_2.4.9-1+1_amd64.deb
[sudo] password for zeddie: 
dpkg: regarding .../network-manager_1.30.0-1_amd64.deb containing network-manager:
 network-manager breaks ppp (<< 2.4.9-1+~)
  ppp (version 2.4.7-2+4.1ubuntu6) is present and installed.

dpkg: error processing archive ./network-manager_1.30.0-1_amd64.deb (--install):
 installing network-manager would break ppp, and
 deconfiguration is not permitted (--auto-deconfigure might help)
(Reading database ... 262191 files and directories currently installed.)
Preparing to unpack .../libgnutls30_3.7.1-1_amd64.deb ...
De-configuring libgnutls30:i386 (3.6.15-4ubuntu2) ...
Unpacking libgnutls30:amd64 (3.7.1-1) over (3.6.15-4ubuntu2) ...
Preparing to unpack ./libnm0_1.30.0-1_amd64.deb ...
Unpacking libnm0:amd64 (1.30.0-1) over (1.26.2-1ubuntu1) ...
Preparing to unpack ./libnettle8_3.7-2.1_amd64.deb ...
De-configuring libnettle8:i386 (3.6-2) ...
Unpacking libnettle8:amd64 (3.7-2.1) over (3.6-2) ...
Preparing to unpack .../network-manager-pptp-gnome_1.2.8-3+b2_amd64.deb ...
Unpacking network-manager-pptp-gnome (1.2.8-3+b2) over (1.2.8-3) ...
Preparing to unpack .../network-manager-pptp_1.2.8-3+b2_amd64.deb ...
Unpacking network-manager-pptp (1.2.8-3+b2) over (1.2.8-3) ...
dpkg: warning: downgrading pptp-linux from 1.10.0-1build1 to 1.10.0-1
Preparing to unpack .../pptp-linux_1.10.0-1_amd64.deb ...
Unpacking pptp-linux (1.10.0-1) over (1.10.0-1build1) ...
Preparing to unpack ./ppp_2.4.9-1+1_amd64.deb ...
Unpacking ppp (2.4.9-1+1) over (2.4.7-2+4.1ubuntu6) ...
dpkg: error processing package libgnutls30:amd64 (--install):
 package libgnutls30:amd64 3.7.1-1 cannot be configured because libgnutls30:i386 is at a different version (3.6.15-4ubuntu2)
dpkg: error processing package libgnutls30:i386 (--install):
 package libgnutls30:i386 3.6.15-4ubuntu2 cannot be configured because libgnutls30:amd64 is at a different version (3.7.1-1)
dpkg: dependency problems prevent configuration of libnm0:amd64:
 libnm0:amd64 depends on libgnutls30 (>= 3.7.0); however:
  Package libgnutls30:amd64 is not configured yet.

dpkg: error processing package libnm0:amd64 (--install):
 dependency problems - leaving unconfigured
dpkg: error processing package libnettle8:amd64 (--install):
 package libnettle8:amd64 3.7-2.1 cannot be configured because libnettle8:i386 is at a different version (3.6-2)
dpkg: error processing package libnettle8:i386 (--install):
 package libnettle8:i386 3.6-2 cannot be configured because libnettle8:amd64 is at a different version (3.7-2.1)
dpkg: dependency problems prevent configuration of network-manager-pptp-gnome:
 network-manager-pptp-gnome depends on libnm0 (>= 1.2.0); however:
  Package libnm0:amd64 is not configured yet.

dpkg: error processing package network-manager-pptp-gnome (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of network-manager-pptp:
 network-manager-pptp depends on libnm0 (>= 1.2.0); however:
  Package libnm0:amd64 is not configured yet.

dpkg: error processing package network-manager-pptp (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of ppp:
 network-manager (1.26.2-1ubuntu1) breaks ppp (>= 2.4.7-3~) and is installed.
  Version of ppp to be configured is 2.4.9-1+1.

dpkg: error processing package ppp (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of pptp-linux:
 pptp-linux depends on ppp; however:
  Package ppp is not configured yet.

dpkg: error processing package pptp-linux (--install):
 dependency problems - leaving unconfigured
Processing triggers for libc-bin (2.32-0ubuntu3) ...
Processing triggers for dbus (1.12.20-1ubuntu1) ...
Processing triggers for man-db (2.9.3-2) ...
Errors were encountered while processing:
 ./network-manager_1.30.0-1_amd64.deb
 libgnutls30:amd64
 libgnutls30:i386
 libnm0:amd64
 libnettle8:amd64
 libnettle8:i386
 network-manager-pptp-gnome
 network-manager-pptp
 ppp
 pptp-linux
`

**Comment by jacobgkau** [comment]
@ZeddieXX First, it looks like you have `libgnutls30:i386` installed. That is not installed by default, only the amd64 version is installed by default. You would either need to remove the i386 version, or download the updated i386 version and include it in the installation command: https://packages.debian.org/sid/i386/libgnutls30/download 

Same thing for libnettle8: https://packages.debian.org/sid/i386/libnettle8/download 

Second, you may need to change the order of packages and/or run the command more than once until all of them install and configure. If I recall, I ran it with `apt install` first, it failed, then I ran it with `dpkg -i` and it succeeded.

**Comment by ZeddieXX** [comment]
> @ZeddieXX First, it looks like you have `libgnutls30:i386` installed. That is not installed by default, only the amd64 version is installed by default. You would either need to remove the i386 version, or download the updated i386 version and include it in the installation command: https://packages.debian.org/sid/i386/libgnutls30/download
> 
> Same thing for libnettle8: https://packages.debian.org/sid/i386/libnettle8/download
> 
> Second, you may need to change the order of packages and/or run the command more than once until all of them install and configure. If I recall, I ran it with `apt install` first, it failed, then I ran it with `dpkg -i` and it succeeded.

I got it installed now by using these two lines:
sudo apt --fix-broken install ./network-manager_1.30.0-1_amd64.deb ./libgnutls30_3.7.1-1_amd64.deb ./libnm0_1.30.0-1_amd64.deb ./libnettle8_3.7-2.1_amd64.deb ./network-manager-pptp-gnome_1.2.8-3+b2_amd64.deb ./network-manager-pptp_1.2.8-3+b2_amd64.deb ./pptp-linux_1.10.0-1_amd64.deb ./ppp_2.4.9-1+1_amd64.deb ./libgnutls30_3.7.1-1_i386.deb ./libnettle8_3.7-2.1_i386.deb
sudo dpkg -i ./network-manager_1.30.0-1_amd64.deb ./libgnutls30_3.7.1-1_amd64.deb ./libnm0_1.30.0-1_amd64.deb ./libnettle8_3.7-2.1_amd64.deb ./network-manager-pptp-gnome_1.2.8-3+b2_amd64.deb ./network-manager-pptp_1.2.8-3+b2_amd64.deb ./pptp-linux_1.10.0-1_amd64.deb ./ppp_2.4.9-1+1_amd64.deb ./libgnutls30_3.7.1-1_i386.deb ./libnettle8_3.7-2.1_i386.deb

**Comment by jacobgkau** [comment]
Glad to hear the newer NetworkManager solves the issue!

> I wonder if I needed the 32-bit packages because I also have Steam/Proton/Lutris installed. I believe they require some 32-bit library for some games.

That would be my guess as well. You could tell for sure by installing aptitude and running `aptitude why libgnutls30:i386`.

> So is it a bug with the Network Manager version that's included in Ubuntu/Pop!OS? Would you push a new version of Network Manager to Pop!OS 20.04 and 20.10?

We're not currently shipping NetworkManager ourselves, it's coming from Ubuntu (you can see this with `apt policy network-manager`.) The link you posted (LP #1565717) was last updated in 2016, so I don't think it's referring to the same issue, but it would make sense to report this to Ubuntu on Launchpad since it also happens in Ubuntu (at least according to my testing; your earlier message indicated otherwise, which I don't entirely understand.)

Like I said, the working version you're running now (1.30.0) will be included by default with Ubuntu/Pop 21.04. I just tested on a 20.04 machine (with NM 1.22.10), and it reconnected ten times in a row (which didn't happen on the 20.10 machine with NM 1.26.2), so this issue doesn't appear to affect that version.

**Comment by ZeddieXX** [comment]
Good to know @jacobgkau .  For now the workaround you gave me works on 2 of my Pop!_OS machines running 20.10.  I hope to upgrade to 21.04 soon!
